sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -263,27 +263,27 @@ Connection getConnection() throws Exception {
     }
 
     public Read<T> withDataSourceConfiguration(DataSourceConfiguration configuration) {
-      checkNotNull(configuration, "configuration");
+      checkNotNull(configuration, "DataSourceConfiguration should be set");
       return toBuilder().setDataSourceConfiguration(configuration).build();
     }
 
     public Read<T> withQuery(String query) {
-      checkNotNull(query, "query");
+      checkNotNull(query, "Query should be set");
```

Comment:
Here, the error is that the user has violated the pluggable type system where only `@Nullable` things are permitted to be `null`. I generally don't perform extra validation for that situation. But it is nice so I am not suggesting that you remove it.

But the message is different that in `validate` because it isn't that the user forgot to set the query, but that they passed an illegal argument.

So I would `checkArgument(query != null, "JdbcIO.withQuery(...) called with null query")` to make it clear that it is the argument that is wrong and tell the user exactly what to do to fix it.